third-party/linshare/src/main/java/org/apache/james/linshare/client/LinshareAPI.java

```diff
@@ -0,0 +1,82 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.linshare.client;
+
+import java.io.File;
+import java.util.List;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+
+import feign.Headers;
+import feign.Param;
+import feign.RequestLine;
+
+public interface LinshareAPI {
+
+    @RequestLine("GET /linshare/webservice/rest/user/v2/documents")
+    @Headers("Accept: application/json")
+    List<Document> listAllDocuments();
+
+    @RequestLine("POST /linshare/webservice/rest/user/v2/documents")
+    @Headers({
+        "Accept: application/json",
+        "Content-Type: multipart/form-data"
```

Comment:
These headers are duplicated every where and should at the very least be in some constants...